### PR TITLE
[BEAM-873] Support For Biq Query 2.0 Sql

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Added support for passing a flag that allows the user to toggle between the two SQL formats supported by BQ. We set the default to True to keep backward compatibility here. 

Tested by running queries in both the formats.

R: @chamikaramj

### CODE REVIEW

File: sdks/python/apache_beam/io/bigquery.py [hunk 1]
```diff
@@ -312,6 +312,10 @@ def __init__(self, table=None, dataset=None, project=None, query=None,
         in a file as a JSON serialized dictionary. This argument needs a value
         only in special cases when returning table rows as dictionaries is not
         desirable.
+      useLegacySql: Specifies whether to use BigQuery's legacy
+        SQL dialect for this query. The default value is true. If set to false,
+        the query will use BigQuery's updated SQL dialect with improved
+        standards compliance. This parameter is forced to True for table inputs.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/bigquery.py]: How about "This parameter is ignored for table inputs." and simply ignore it.

File: sdks/python/apache_beam/runners/dataflow_runner.py [hunk 2]
```diff
@@ -515,6 +515,8 @@ def run_Read(self, transform_node):
                             transform.source.table_reference.projectId)
       elif transform.source.query is not None:
         step.add_property(PropertyNames.BIGQUERY_QUERY, transform.source.query)
+        step.add_property(PropertyNames.BIGQUERY_USE_LEGACY_SQL,
+                          transform.source.use_legacy_sql)
```
Comment (chamikaramj) [inline sdks/python/apache_beam/runners/dataflow_runner.py]: Did you test this using DataflowRunner ?

File: sdks/python/apache_beam/io/bigquery.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/bigquery.py]: Will do.

File: sdks/python/apache_beam/utils/names.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ class PropertyNames(object):
   BIGQUERY_CREATE_DISPOSITION = 'create_disposition'
   BIGQUERY_DATASET = 'dataset'
   BIGQUERY_QUERY = 'bigquery_query'
+  BIGQUERY_USE_LEGACY_SQL = "bigquery_use_legacy_sql"
```
Comment (chamikaramj) [inline sdks/python/apache_beam/utils/names.py]: nit: use ' instead of "

**Comment by chamikaramj** [comment]
LGTM

**Comment by sb2nov** [comment]
cc @robertwb for merging

**Comment by robertwb** [comment]
The Jenkins failure is unrelated (java). Merging.

**Comment by sb2nov** [comment]
Thanks

**Pull request closed without merging**